src/Microsoft.Health.Fhir.CosmosDb/Registration/FhirServerBuilderCosmosDbRegistrationExtensions.cs

```diff
@@ -271,6 +272,10 @@ private static IFhirServerBuilder AddCosmosDbPersistence(this IFhirServerBuilder
                 .AsImplementedInterfaces()
                 .AsFactory<IScoped<IQueueClient>>();
 
+            services.Add<CosmosAccessTokenProviderFactory>(c => c.GetRequiredService<IAccessTokenProvider>)
```

Comment:
I think we need one more change for oss, we probably need to always register AzureAccessTokenProvider else this won't work if storage isn't setup

https://github.com/microsoft/fhir-server/blob/e891ceafb2f967110cb4df846945736fcbe3d1b5/src/Microsoft.Health.Fhir.Azure/FhirServerBuilderAzureRegistrationExtensions.cs#L49-L58